src/app/archive/archive_lib/processor.ml

```diff
@@ -1066,6 +1066,79 @@ module Block = struct
          |sql})
       id
 
+  let delete (module Conn : CONNECTION) ~(state_hash : State_hash.t) =
+    let open Deferred.Result.Let_syntax in
+    let state_hash_base58 = State_hash.to_base58_check state_hash in
+    (* delete user commands *)
+    let%bind () =
+      Conn.exec
+        (Caqti_request.exec Caqti_type.string
+           {sql| DELETE FROM user_commands
+                 WHERE id IN
+                  (SELECT user_command_id FROM blocks_user_commands
+                   INNER JOIN blocks
+                   ON blocks.id = block_id
+                   WHERE blocks.state_hash = ?)
+           |sql})
+        state_hash_base58
+    in
+    (* delete internal commands *)
+    let%bind () =
+      Conn.exec
+        (Caqti_request.exec Caqti_type.string
+           {sql| DELETE FROM internal_commands
```

Comment:
I don't think we want to delete internal commands, since they can be shared between multiple blocks. (I also think cascade may delete all of these blocks when we do so.)